### ISSUE TITLE
ASkyBlockAPI#getIslandAt throwing NullPointerException

### ISSUE THREAD

**Issue by NxDs** [tastybento/ASkyBlock-Bugs-N-Features#800]
Thank you for filing a bug report. Please complete these sections to help speed resolution.

**Server version? Remember to say if it is Spigot/Bukkit/Paper etc.**
TacoSpigot 1.8
**Plugin version:**
3.0.8.8
**Is this a new install or upgrade from earlier plugin version? If upgrade, what version did you use before?**
3.0.8.6
**(Optional) What other plugins are you using now? (do /plugins and paste text from the console log here)**

**Description of issue. What happened?**
Getting NullPointerException while accessing the API
**Steps to make this happen**
Call api.getIslandAt(location) where location is not an island or it's spawn
Example of the code currently causing the issue:
```    
public void updateWorldBorder(Player player) {
        if (!settings.isIslandWorldBorderEnabled()) return;

        Island island = aSkyBlockAPI.getIslandAt(player.getLocation());
}
```
Method called after a PlayerTeleportEvent.

Exception thrown:
```
[22:39:58 ERROR]: Could not pass event PlayerTeleportEvent to ASkyBlockPlus v1.0
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader.lambda$0(JavaPluginLoader.java:302) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer.teleport(CraftPlayer.java:530) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at usermanager.spigot.Teleporter.teleportPlayer(Teleporter.java:263) ~[?:?]
        at usermanager.spigot.Teleporter.lambda$teleportPlayerToLocation$1(Teleporter.java:89) ~[?:?]
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:67) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:364) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:797) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:376) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:726) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:629) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.NullPointerException
        at com.wasteofplastic.askyblock.Island.<init>(Island.java:472) ~[?:?]
        at com.wasteofplastic.askyblock.ASkyBlockAPI.getIslandAt(ASkyBlockAPI.java:460) ~[?:?]
        at net.extremecraft.askyblockplus.listeners.SkyblockListener.onPlayerTeleportEventLowest(SkyblockListener.java:162) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor131.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
        at org.bukkit.plugin.java.JavaPluginLoader.lambda$0(JavaPluginLoader.java:300) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        ... 14 more
[22:39:58 WARN]: [ASkyBlockPlus] Task #2171 for ASkyBlockPlus v1.0 generated an exception
java.lang.NullPointerException
        at com.wasteofplastic.askyblock.Island.<init>(Island.java:472) ~[?:?]
        at com.wasteofplastic.askyblock.ASkyBlockAPI.getIslandAt(ASkyBlockAPI.java:460) ~[?:?]
        at net.extremecraft.askyblockplus.ASkyblockPlus.updateWorldBorder(ASkyblockPlus.java:375) ~[?:?]
        at net.extremecraft.askyblockplus.listeners.SkyblockListener.lambda$onPlayerTeleportEventMonitor$2(SkyblockListener.java:184) ~[?:?]
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:67) ~[spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:364) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:797) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:376) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:726) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:629) [spigot-dev3.jar:git-TacoSpigot-7eab5bf6]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]

```

It works perfectly fine with the version 3.0.8.6
**What do you think should happen?**
It shouldn't be throwing the NPE, at best it should return null

**Comment by tastybento** [comment]
This is fixed in commit 3317726373e9f4729b9458263e9c5411838b0d9d.
It'll be in the next release.

**Comment by NxDs** [comment]
I can confirm that version fixes
